pkg/tools/prometheus/server.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2024 Nordix Foundation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package prometheus provides a set of utilities for assisting with Prometheus data
+package prometheus
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+)
+
```

Comment:
Since we plan to use it in all repos where we can grab metrics at this moment (forwarder ovs/sriov/vpp), I think it's important to move the logic with listen and serve to this package to make it easier to update:


```suggestion
func ListenAndServe(ctx context.Context, listenOn string) {

metricsServer := prometheus.Server{
			IP:            cfg.PrometheusIP,
			Port:          int(cfg.PrometheusPort),
			HeaderTimeout: cfg.PrometheusServerHeaderTimeout,
		}

		tlsConfig := &tls.Config{
			MinVersion: tls.VersionTLS12,
		}
		source, err := workloadapi.NewX509Source(ctx)
		if err != nil {
			log.FromContext(ctx).Fatalf("error getting x509 source: %v", err.Error())
		}
		tlsConfig.GetCertificate = tlsconfig.GetCertificate(source)
		metricsServer.TLSConfig = tlsConfig

		select {
		case <-ctx.Done():
			err = source.Close()
			log.FromContext(ctx).Errorf("unable to close x509 source: %v", err.Error())
		default:
		}

		
		err := metricsServer.Start(ctx)
		if err != nil {
			log.FromContext(ctx).Error(err.Error())
			cancel()
		}
}
```

Then we'll able to use this in other repos:
```go
if prometheus.IsEnabled() {
   go prometheus.ListenAndServe(ctx, listenOn, ...)
}
```